### PR TITLE
Rename ``useless-super-delegation`` to ``useless-parent-delegation``

### LINKED ISSUE #6953: Rename ``useless-super-delegation`` to ``useless-parent-delegation``
### Current problem

Ref. https://github.com/PyCQA/pylint/issues/5083

It makes sense to rename these as we could then extend the message a bit.

### Desired solution

See title.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Pierre-Sassoulas**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |
| ✓   | :sparkles: New feature |
| ✓   | :hammer: Refactoring   |
| ✓   | :scroll: Docs          |

## Description

Closes #6953

### CODE REVIEW

File: pylint/checkers/classes/class_checker.py [hunk 1]
```diff
@@ -579,12 +579,13 @@ def _has_same_layout_slots(slots, assigned_value):
...
-        "Useless super delegation in method %r",
-        "useless-super-delegation",
+    "W0246": (
+        "Useless parent delegation in method %r",
+        "useless-parent-delegation",
         "Used whenever we can detect that an overridden method is useless, "
-        "relying on super() delegation to do the same thing as another method "
+        "relying on parent delegation to do the same thing as another method "
         "from the MRO.",
+        {"old_names": [("W0235", "useless-super-delegation")]},
```
Comment (Pierre-Sassoulas) [inline pylint/checkers/classes/class_checker.py:588]: Using a tuple here trip up the check for ``consider-using-namedtuple-or-dataclass`` for the whole dict.

File: pylint/checkers/classes/class_checker.py [hunk 2]
```diff
@@ -579,12 +577,13 @@ def _has_same_layout_slots(slots, assigned_value):
         "Used when an __init__ method is called on a class which is not "
         "in the direct ancestors for the analysed class.",
     ),
-    "W0235": (
-        "Useless super delegation in method %r",
-        "useless-super-delegation",
+    "W0246": (
+        "Useless parent delegation in method %r",
```
Comment (DanielNoord) [inline pylint/checkers/classes/class_checker.py:581]:
```suggestion
        "Useless parent or super() delegation in method %r",
```

File: pylint/checkers/classes/class_checker.py [hunk 3]
```diff
@@ -579,12 +577,13 @@ def _has_same_layout_slots(slots, assigned_value):
...
+        "useless-parent-delegation",
         "Used whenever we can detect that an overridden method is useless, "
-        "relying on super() delegation to do the same thing as another method "
+        "relying on parent delegation to do the same thing as another method "
```
Comment (DanielNoord) [inline pylint/checkers/classes/class_checker.py:584]:
```suggestion
        "relying on parent or super() delegation to do the same thing as another method "
```

File: tests/functional/u/useless/useless_super_delegation.py [hunk 4]
```diff
@@ -211,63 +211,63 @@ def with_default_unhandled(self, first, default_arg=lambda: True):
 
 class UselessSuper(Base):
 
-    def equivalent_params(self): # [useless-super-delegation]
+    def equivalent_params(self): # [useless-parent-delegation]
```
Comment (DanielNoord) [inline tests/functional/u/useless/useless_super_delegation.py:214]: We should probably rename all test files.

**Review by DanielNoord — CHANGES_REQUESTED** [review]

**Review by DanielNoord — APPROVED** [review]

**Pull request merged**